src/fondant/pipeline/lightweight_component.py

```diff
@@ -65,11 +68,62 @@ class LightweightComponent(BaseComponent):
     def image(cls) -> Image:
         raise NotImplementedError
 
+    @classmethod
+    def consumes(cls) -> t.Optional[t.Dict[str, t.Any]]:
+        pass
+
+    @classmethod
+    def modify_consumes_spec(cls, apply_consumes, consumes_spec):
+        """Modify fields based on the consumes argument in the 'apply' method."""
+        if apply_consumes:
+            for k, v in apply_consumes.items():
+                if isinstance(v, str):
+                    consumes_spec[k] = consumes_spec.pop(v)
+                else:
+                    msg = (
+                        f"Invalid data type for field `{k}` in the `apply_consumes` "
+                        f"argument. Only string and pa.DataType are allowed."
+                    )
+                    raise ValueError(
+                        msg,
+                    )
+        return consumes_spec
+
+    @classmethod
+    def get_consumes_spec(
+        cls,
+        dataset_fields: t.Mapping[str, Field],
+        apply_consumes: t.Optional[t.Dict[str, t.Union[str, pa.DataType]]] = None,
+    ):
+        consumes = cls.consumes()
+
+        if consumes is None:
+            # Get consumes spec from the dataset
+            spec_consumes = {k: v.type.to_dict() for k, v in dataset_fields.items()}
+
+            spec_consumes = cls.modify_consumes_spec(apply_consumes, spec_consumes)
```

Comment:
Nitpick: I would use consistent naming, so either `consumes_spec` or `spec_consumes`.
Maybe we can use inside the  `modify_consumes_spec` something like `dataset_consumes` to make even more clear.